client/views/room/contextualBar/Call/Jitsi/CallJitsWithData.js

```diff
@@ -119,43 +124,76 @@ const CallJitsWithData = ({ rid }) => {
 		user.username,
 	]);
 
-	const testAndHandleTimeout = useMutableCallback(() => {
+	const testAndHandleTimeout = useMutableCallback(async () => {
 		if (jitsi.openNewWindow) {
 			if (jitsi.window?.closed) {
 				return jitsi.dispose();
 			}
-			return updateTimeout(rid, false);
+			try {
+				await updateTimeout(rid, false);
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: t(error.reason) });
+				clear();
+				handleClose();
+				return jitsi.dispose();
+			}
 		}
 		if (new Date() - new Date(room.jitsiTimeout) > TIMEOUT) {
 			return jitsi.dispose();
 		}
 
 		if (new Date() - new Date(room.jitsiTimeout) + TIMEOUT > DEBOUNCE) {
-			return updateTimeout(rid, false);
+			try {
+				await updateTimeout(rid, false);
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: t(error.reason) });
+				clear();
+				handleClose();
+				return jitsi.dispose();
+			}
 		}
 	});
 
 	useEffect(() => {
-		if (!accepted || !jitsi) {
-			return;
-		}
+		async function fetchData() {
+			if (!accepted || !jitsi) {
+				return;
+			}
 
-		if (jitsi.needsStart) {
-			jitsi.start(ref.current);
-			updateTimeout(rid, true);
-		} else {
-			updateTimeout(rid, false);
-		}
+			const none = () => {};
+			const clear = () => {
+				jitsi.off('HEARTBEAT', testAndHandleTimeout);
+				jitsi.dispose();
+			};
+
+			try {
+				if (jitsi.needsStart) {
+					jitsi.start(ref.current);
+					await updateTimeout(rid, true);
+				} else {
+					await updateTimeout(rid, false);
+				}
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: t(error.reason) });
+				clear();
+				handleClose();
+			}
 
-		jitsi.on('HEARTBEAT', testAndHandleTimeout);
-		const none = () => {};
-		const clear = () => {
-			jitsi.off('HEARTBEAT', testAndHandleTimeout);
-			jitsi.dispose();
-		};
+			jitsi.on('HEARTBEAT', testAndHandleTimeout);
 
-		return jitsi.openNewWindow ? none : clear;
-	}, [accepted, jitsi, rid, testAndHandleTimeout, updateTimeout]);
+			return jitsi.openNewWindow ? none : clear;
+		}
+		return fetchData();
```

Comment:
**You cannot return a `Promise` object from `useEffect` callback.** In this case, you are returning a promise of type `Promise<() => void>`; it should be a clean up function of type `() => void`.
If you need to await for the `updateTimeout` to decide for calling `jitsi.dispose()` or not on effect clean up, consider using a local boolean variable. E.g.:

```
useEffect(() => {
  let shoudDispose = false;
  
  async function fetchData() {
     ...
     shouldDispose = !jitsi.openNewWindow;
  }
  
  fetchData();
  
  return () => {
    if (shouldDispose) {
      jitsi.dispose();
      // whatever more
    }
  };
}, [...]);
```